src/main/vcpf4/usbd_cdc_vcp.c

```diff
@@ -165,10 +163,15 @@ uint32_t CDC_Send_DATA(uint8_t *ptrBuffer, uint8_t sendLength)
     return sendLength;
 }
 
+uint32_t CDC_Send_FreeBytes(void)
+{
+	/* return the bytes free in the circular buffer */
+    return ((APP_Rx_ptr_out - APP_Rx_ptr_in) + (-((int)(APP_Rx_ptr_out <= APP_Rx_ptr_in)) & APP_RX_DATA_SIZE)) - 1;
```

Comment:
I don't think it actually is. The compiler can inline variables as it likes. Even if it wasn't exactly the fastest I think readability is more important.